lib/wasix/src/bin_factory/binary_package.rs

```diff
@@ -90,7 +121,9 @@ impl BinaryPackage {
 
         let manifest = container.manifest();
         let id = PackageInfo::package_id_from_manifest(manifest)?.unwrap_or_else(|| {
-            PackageId::Hash(PackageHash::from_sha256_bytes(container.webc_hash()))
+            PackageId::Hash(PackageHash::from_sha256_bytes(
+                container.webc_hash().unwrap(),
```

Comment:
I'd use `container.webc_hash().unwrap_or(|| container.path_hash())` ?

Reply:
calling `from_webc` with a `Package`, is a bug (or not intended usage) from this PR forward since the hash won't be present. Although your suggestion works, I prefer the dedicated `from_dir` approach since it clearly signals the difference between a local package and a local webc.

To not panic, I changed the code to return an error upon the absence of the webc hash.